bundles/zig/zig_lexer.moon

```diff
@@ -0,0 +1,133 @@
+-- Copyright 2020 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+howl.util.lpeg_lexer ->
+  c = capture
+
+  -- Shorthand for lexer.word
+  ident = (alpha + '_')^1 * (alpha + digit + '_')^0
+
+  -- Comments
+  line_comment = P'//' * scan_until eol
+  comment = c 'comment', line_comment
+
+  -- Hex digits
+  hex_digit = R'09' + R'af' + R'AF'
+
+  -- Strings
+  dq_str = span '"', '"', '\\'
+  cont = R'\128\191'
+  utf8 = R'\0\127' + R'\194\223' * cont + R'\224\239' * cont * cont + R'\240\244' * cont * cont * cont
+  ascii_esc = '\\' * S'trn\'"\\0'
+  unicode_esc = ('\\u{' * hex_digit^1 * '}')
+  hex_esc = ('\\x' * hex_digit^1)
+  char = P"'" * (unicode_esc + hex_esc + ascii_esc + utf8) * P"'"
+  string  = c 'string', any {dq_str, char}
+
+  -- Multi-line strings
+  multi_span = P'\\\\' * scan_until eol
+  multi_string = c 'string', multi_span
+
+  -- Numbers
+  binary = P'0b' * S'01'^1
+  oct = P'0o' * S'01234567'^1
+  hex = P'0x' * hex_digit^1
+  decimal = digit^1
+  floats = float * (S'eE' * S'+-'^-1 * decimal)^-1
+  number = c 'number', any {
+    binary,
+    hex,
+    oct,
+    decimal,
+    floats
+  }
+
+  -- Keywords
+  keyword = c 'keyword', word {
+    'align', 'allowzero', 'and', 'anyframe', 'asm', 'async', 'await', 'break',
+    'callconv', 'catch', 'comptime', 'const', 'continue', 'defer', 'else',
+    'enum', 'errdefer', 'error', 'export', 'extern', 'fn', 'for', 'if',
+    'inline', 'noalias', 'orelse', 'or', 'packed', 'promise', 'pub', 'resume',
+    'return', 'linksection', 'struct', 'suspend', 'switch', 'test',
+    'threadlocal', 'try', 'union', 'unreachable', 'usingnamespace', 'var',
+    'volatile', 'while'
+  }
+
+  -- Special words
+  special = c 'special', word {
+      'true', 'false', 'null', 'undefined', '@addWithOverflow', '@alignCast',
```

Comment:
I'm not intimately familiar with Zig, but if all these start with an `@` sign, then I'm guessing that signifies some sort of compiler-level built-in. Could it then just match any word after an `@` as special? Might be also worth handling a plain `@` then too so it looks prettier as you type the code.

(As a side note, builtin *functions* are usually lexed using the `function` color, and *compiler directives* are usually `operator` [I think...]. I'm not sure where these fall exactly, but it may be worth taking a look at those two color options as well.)

Reply:
`@` is supposed to be used for builtin functions. The language documentation's own highlighting has it distinct as its own color and its been tokenized as 'builtin'. I'm open to any suggestion how else it should be tagged, but I do think they should remain distinctly colored the way they are in the docs. 

As to your question about matching any word, I wouldn't do that in the general case as it quickly offers feedback right now when you've got the builtin correct. I think I would miss not having that as a feature. I constructed that list with a shell one-liner scraping the docs page thanks to the CSS tags including the token type. I don't imagine they'd have to be updated frequently, either.

However, one thing unsupported but related is the `@""` syntax for treating the quoted string as a string token, rather than a keyword or random bytes, with  `foo.@“bar”` being identical to foo.bar. This allows for you to do things like `@"😁"` as in `pub fn @"😁"(self: @This()) void { }` or `export fn @"A function name that is a complete sentence."() void {}`. There's no rule for matching that as anything other a bare `@` followed by a string. That seemed fine enough.

Reply:
Ah okay, so in that case you should probably use the `function` color for these, which is what builtins generally use in other languages (see Python for an example). It's hard to compare syntax highlighting across editors / libraries / sites, because they all tend to separate the different colors differently.

Reply:
I personally will not be doing that on my end, but if that's what you wish to do in order to merge the PR, that's fine.